src/components/Atoms/Input/index.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef } from "react";
+import styled from "styled-components";
+import COLOR from "../../../variables/color";
+import TEXT from "../../../variables/texts";
+
+export const Input = ({ onEditComplete, defaultValue = "" }) => {
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
+
+  const handleBlur = () => {
+    onEditComplete(inputRef.current.value);
+  };
+
+  const handleKeyDown = (e) => {
+    if (e.key !== "Enter") return;
+    onEditComplete(inputRef.current.value);
+  };
+
+  return (
+    <StyledInput
+      defaultValue={defaultValue}
+      onBlur={handleBlur}
+      onKeyDown={handleKeyDown}
+      ref={inputRef}
+    />
+  );
+};
+
+const StyledInput = styled.input.attrs((props) => ({ ...props }))`
```

Comment:
このあたり特にいらない気もしますがいかがでしょう？
```suggestion
const StyledInput = styled.input`
```

Reply:
コメントくださりありがとうどざいます。最初にdivを設定してしまい上手くいかなかったため他の方のInputを参考にして後ろのコードを書きました。調べたところhtml属性を追加していることは分かったのですが、なくしても実装できました。html属性を指定しなくても大丈夫でしょうか。

Reply:
> html属性を指定しなくても大丈夫でしょうか。

ここについては、JSXにhtml属性に対応する属性を指定すれば attrs メソッドを書かなくてもhtml属性は渡されるので心配無用です。

---

`.attrs` は、styled-component に渡ってきたPropsを加工して渡すためのメソッドです。
使い方については[公式ドキュメント](https://styled-components.com/docs/api#.attrs)がわかりやすいです。

`.attrs((props) => ({ ...props }))`
と
`.attrs((props) => (props))`
が意味するところは同じで、styled-component に渡されたPropsをそのままDOM要素（HTML要素）に渡しています。結果的には、attrs メソッドを呼ばない場合と全く同じに鳴ります。

Reply:
コメントくださりありがとうございます。
今回は属性指定なしバージョンをマージしたいと思います。
詳しく教えてくださりありがとうございました。